### ISSUE TITLE
Allow disabling diagnostics

### ISSUE THREAD

**Issue by plexus** [BetterThanTomorrow/calva#980]
Calva receives various diagnostics from clojure-lsp (which gets most but not all of these from clj-kondo), it then displays them as squiggly annotations.

Not everyone wants this, I mostly perceive these as noise. People with ADHD in particular may find these really troublesome.

I could not find a way to disable these. Calva and clojure-lsp don't seem to have a flag for it. I couldn't find a way to tell VS Code to ignore linting/annotations. The best you can do right now is to disable linters in clj-kondo, but not all diagnostics are coming from clj-kondo (e.g. unused namespace, unused public are coming from clojure-lsp itself).

I'm testing out a patch that adds a configuration flag for this, essentially this:

``` json
// package.json
                    "calva.displayDiagnostics": {
                        "type": "boolean",
                        "markdownDescription": "Show linter warnings and other diagnostics from clojure-lsp and clj-kondo.",
                        "default": true
                    }
```

``` typescript
// lsp.ts
        middleware: {
            handleDiagnostics(uri, diagnostics, next) {
                if (!state.config().displayDiagnostics) {
                    return;
                }
```

If it works I can send a PR, meanwhile happy to get some input on this. Is this a reasonable place to handle this? Any opinions on the config flag / description?

**Comment by bpringe** [comment]
Hey @plexus. Thanks for taking time to look into this. That is the appropriate solution, but I've not added that because we have been waiting on https://github.com/clojure-lsp/clojure-lsp/pull/261 to be merged, which I believe uses only clj-kondo for diagnostics, in which case disabling clj-kondo diagnostics as per clj-kondo's instructions will/should work, and this Calva toggle is superfluous, imo.

However, it may still be a good idea to add this setting? @PEZ Thoughts? 

So if you disable all linting via kondo's config, I think it may cause kondo to not do the linting work in the first place (@borkdude correct me if I'm wrong). But with this, the linting work is done, just not shown to the user. So what I mean to say is if that's true, this may not be a good setting to provide as there's a better way to disable diagnostics.

**Comment by bpringe** [comment]
That PR is complete now, so we just need to merge into Calva and do some testing. Once we release then the disabling of all linting via kondo's config should work, if I understand correctly.
